app/models/topic.rb

```diff
@@ -195,6 +195,8 @@ def recover!(recovered_by = nil)
     end
   end
 
+  validates :external_id, allow_nil: true, uniqueness: true, length: { maximum: 50 }, format: { with: /\A[\w-]+\z/ }
```

Comment:
I think we can move `50` to a constant so that it can be reused in the test cases.